docs/source/references/network_events.rst

```diff
@@ -0,0 +1,200 @@
+    ...
+
+
+Event Filters
+-------------
+
+To aid with query limits and more specific queries, events can be filtered using the ``--event-filter``
+option. Multiple ``--event-filter`` options can be defined, but note that only properties classified
+as ``indexed`` in the event's solidity definition can be used as a filter.
+
+For example, to view all of the commitments ever made by the Worker associated with a specific Staker run:
+
+.. code::
+
+    $ nucypher status events --provider <PROVIDER URI> --contract-name StakingEscrow --event-name CommitmentMade --event-filter staker=<STAKING_ADDRESS> --from-block 0
+
+    Reading Latest Chaindata...
+    Retrieving events from block 0 to latest
+
+    --------- StakingEscrow Events ---------
+
+    CommitmentMade:
+      - (EventRecord) staker: <STAKER_ADDRESS>, period: 18551, value: 1234567890123456789012, block_number: 11057641
+      - (EventRecord) staker: <STAKER_ADDRESS>, period: 18552, value: 1234567890123456789012, block_number: 11063640
+      - (EventRecord) staker: <STAKER_ADDRESS>, period: 18553, value: 1234567890123456789012, block_number: 11070103
+      - (EventRecord) staker: <STAKER_ADDRESS>, period: 18554, value: 1234567890123456789012, block_number: 11076964
+      ...
+
+To view the commitment made by the Worker associated with a specific Staker in period 18552, run:
+
+.. code::
+
+    $ nucypher status events --provider <PROVIDER URI> --contract-name StakingEscrow --event-name CommitmentMade --event-filter staker=<STAKING_ADDRESS> --event-filter period=18552 --from-block 0
+
+    Reading Latest Chaindata...
+    Retrieving events from block 0 to latest
+
+    --------- StakingEscrow Events ---------
+
+    CommitmentMade:
+      - (EventRecord) staker: <STAKER_ADDRESS>, period: 18552, value: 1234567890123456789012, block_number: 11063640
+
+
+CSV Output
+----------
+
+CLI output can be cumbersome when trying to generate insights and correlate different events. Instead, the event
+data can be written to a CSV file using either of the following command-line options:
+
+* ``--csv`` - flag to write event information to default CSV files in the current directory with default filenames
+* ``--csv-file <FILEPATH>`` - write event information to a specific CSV file at the provided filepath
+
+
+For example,
+
+.. code::
+
+    $ nucypher status events --provider <PROVIDER URI> --contract-name PolicyManager --event-name PolicyCreated --from-block 0 --csv
+
+    Reading Latest Chaindata...
+    Retrieving events from block 0 to latest
+
+    --------- PolicyManager Events ---------
+
+    PolicyManager::PolicyCreated events written to PolicyManager_PolicyCreated_2021-02-24_20-02-32.csv
+
+
+.. code::
+
+    $ nucypher status events --provider <PROVIDER URI> --contract-name PolicyManager --event-name PolicyCreated --from-block 0 --csv-file ~/Policy_Events.csv
```

Comment:
Expanding a bit more, listing several latest events for a given staker when using `nucypher status stakers --staking-address ...` would be helpful for individual node operators who monitor their stake status using the CLI.

Reply:
✔️